test/hotspot/jtreg/compiler/arguments/TestManyParameters.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @requires os.arch=="amd64" | os.arch=="x86_64"
+ * @bug 8342156
+ * @summary Check that C2 restriction on number of method arguments is not too
+ *          restrictive on x64.
+ *
+ * @run main/othervm -Xcomp
+ *                   -XX:CompileCommand=compileonly,compiler.arguments.TestManyParameters::test
+ *                   -XX:+UnlockDiagnosticVMOptions
+ *                   -XX:+AbortVMOnCompilationFailure
+ *                   compiler.arguments.TestManyParameters
+ */
+
+package compiler.arguments;
+
+public class TestManyParameters {
+
+    public static void main(String[] args) {
+        test(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54);
```

Comment:
Is this the maximum number of arguments that C2 could handle before the addition of APX?

Reply:
Yes, that's my understanding (more details are in the JBS issue). @chhagedorn: Can you confirm?

Reply:
Yes, exactly.

Reply:
OK, thanks!